### ISSUE TITLE
Official Support for MOLPay (including MOLPay CASH) and Billplz

### ISSUE THREAD

**Issue by Azyhan96** [snipcart/snipcart-v3-feedback#13]
# Spoiler alert: This feature request is focused on Malaysian market (may also be applied to neighboring country)

### I request for this payment gateway to have an official support because the current available payment gateway supported by snipcart, either it is not supported yet in our country or the rate is quite pricey in Malaysian market. Take braintree for example, US market it is priced at 2.9% + $0.30 meanwhile when entering Malaysian market it is priced at 3.40% + RM 2.00 . Yes there will always be a cost in eCommerce, but merchant will opt to those that cost less or at least reasonable for them to pay. 

> other payment gateways support will be **top priority** during the beta phase. 
Source: https://snipcart-docs-v3.netlify.com/cart/v3/installation

---
name: **Feature request**
about: _Support for MOLPay (including MOLPay CASH) and Billplz_
title: **Support for Malaysian Payment Gateway**
labels: **enhancement**
assignees: ''

---

**Is your feature request related to a problem? Please describe.**
As far as i know, current payment method supported by Snipcart is only via card. While it is not denied that is the fastest way to receive payment from buyer, there is a few problem worth to mention. **Paying by card is not preferred method in Malaysia**. There is a few factor that cause this issue:
1. Local bank block purchasing online using debit card **by default**. Whoever want to unblock the feature **must** call bank representative (no other way available). **FPX** is most preferred method here because any transaction that will be made required specially generated **TAC** number.
2.  Card also have the above feature but it is known as **3d secure**. As far as i know, it is not supported by all bank yet (correct me if i'm wrong) . If all of them already does, the current problem is Snipcart still didn't support 3d secure (**as far as i remember**)
3. There is also a minority, but worth attention, type of online buyer. The one who prefer using cash (**cash is still king**). I know that there is current supported method by Snipcart called `deferred payment` . So far it can only represent Cash On Delivery and Bank In at any available Cash Deposit Machine. Meanwhile there is already modern type of paying by cash option available here which is pay by cash at any local 7e store.

**Describe the solution you'd like**
The list below will reflect list above:
1. This issue can be solved if there is an **official support for MOLPay and Billplz** by Snipcart. Both accept FPX as the payment method. The only difference is the rate they offer. Billplz is known for lower rate while MOLPay is known for MOLPay CASH (preferred by someone who is new to online shopping)
2. **Official support by snipcart for 3d secure.**
3. **Official support for MOLPay CASH**. As mentioned above, it is preferred by someone who is new to online shopping. Cash on Delivery is good, the problem is, not all courier accept this and not all merchant know how to opt for this feature offered by courier. Bank In at any local Cash Deposit Machine (CDM) is good, but the problem is, it is hard to find CDM for the bank account provided by merchant (this explain why some merchant have a lots of different bank account, just the fact that some CDM is hard to find at some place). 

**Describe alternatives you've considered**
Provide a way to support custom payment gateway, with guide from snipcart team if possible. Anyway, **official support for above mentioned payment gateway is preferred** as it will be up to date with current version of the above payment method and, code wise, **Snipcart team code better**. (I mention this because i once encounter an issue with payment on local eCommerce site, tl;dr payment made not detected by merchant's site. So i need to issue a refund with evidence that it is already being deducted from my bank account)

**Additional context**
Link to the above mentioned payment gateway
MOLPay : https://www.molpay.com/v3/
Billplz: https://www.billplz.com/

**Additional Info**
Most of eCommerce site here actually accept FPX as payment method which support my statement that it is preferred payment method here. For example [lazada](https://www.lazada.com.my/) . About bank block the card functionality for online payment, some bank offer _safer_ alternative to allow that, guess what? A [prepaid card](https://www.cimbbank.com.my/en/personal/products/cards/view-all-prepaid-cards/cimb-lazada-prepaid-mastercard.html) . Even local airline group here also start promoting their prepaid card called [BigPay](https://www.bigpayme.com/) . There is also eWallet rising here such as Boost and GrabPay. Guess what? **All mentioned above actually support FPX**. Why? As i said, it is most preferred payment method here. ¯\\\_(ツ)_/¯

## While writing this, i didn't know how many local developer have Snipcart in their Arsenal, but i believe, by having this feature, it will increase the number. Most of local developer here use WooCommerce, but there is already a few developer opt for alternative solution. 

**Also worth to mention here, i still can't test snipcart v3 beta because Stripe isn't available yet in my country. I will test when fake payment gateway is available.**

**Edit**: Update some explanation and add some content to support this statement.
**Edit** _again_ : Update shrug XD

**Comment by JeanSebTr** [comment]
Thanks for the extensive explanation @Azyhan96! 🙂 

While the V3 is mostly focussed on the Cart, we did revamp our payment processing system from the ground up. This new system will make it easier for us to implement new gateways or to support alternative payment methods to credit cards.

We received a few requests for MOLPay and Billplz but they aren't yet on our roadmap. However, we plan to offer a way to implement custom payment gateways.

As for 3D Secure, all new V3 gateway integrations will be compatible. This is required for PSD2 before September 14.

**Comment by Azyhan96** [comment]
update:

stripe now already available in Malaysia but still in beta.
i ask them and they said that they already doing some works to support FPX.

since Snipcart support Stripe by default it is a good news but bad news is that the cost is quite high compare to local payment gateway.

anyway, will wait for feature to integrate other payment gateway.

more option, the better.
